### PR TITLE
feat: add toke reward claim functionality

### PR DESCRIPTION
**Pull request by toshiSat**
- Move all abi's into their own folder.  Started getting a lot of them
- Add canSendWithdrawalRequest & canClaimWithdraw functions
- Add toke reward claim functions

### CODE REVIEW

File: packages/investor-foxy/src/api/api.ts [hunk 1]
```diff
@@ -805,6 +975,103 @@ export class FoxyApi {
...
+      try {
+        return rewardHashContract.methods.cycleHashes(latestCycleIndex).call()
+      } catch (e) {
+        throw new Error(`Failed to get latestCycleIndex, ${e}`)
+      }
+    })()
+
+    try {
+      const { latestClaimable } = cycleHashes
+      const response = await fetch(
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:998]: axios<AddResponseTypeHere>

File: packages/investor-foxy/src/api/api.ts [hunk 2]
```diff
@@ -805,6 +975,103 @@ export class FoxyApi {
...
+    }
+  }
+
+  async claimFromTokemak(input: TxInput): Promise<string> {
+    const { contractAddress, wallet, userAddress, accountNumber = 0, dryRun = false } = input
+    this.verifyAddresses([contractAddress])
+
+    const { latestCycleIndex, claimAmount, v, r, s } = await this.getTokeRewardAmount(input)
+
+    if (!wallet) throw new Error('Missing inputs')
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:1025]: move this up to 1020 and fail faster

File: packages/investor-foxy/src/api/api.ts [hunk 3]
```diff
@@ -805,6 +975,103 @@ export class FoxyApi {
...
+    }
+
+    const recipient = {
+      chainId: 1,
+      cycle: latestCycleIndex,
+      wallet: contractAddress,
+      amount: claimAmount
+    }
+
+    const estimatedGasBN = (async () => {
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:1043]: await?

File: packages/investor-foxy/src/api/api.ts [hunk 4]
```diff
@@ -551,6 +656,76 @@ export class FoxyApi {
...
+        return 0
+      }
+    })()
+
+    const nextCycleStart = bnOrZero(currentCycleStart).plus(duration)
+
+    const blockNumber = await this.web3.eth.getBlockNumber()
+    const timestamp = (await this.web3.eth.getBlock(blockNumber)).timestamp
+
+    const isTimeToRequest = timestamp + timeLeftToRequestWithdrawal >= nextCycleStart
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:722]: bn

**Review by 0xdef1cafe — APPROVED** [review]

**Review by DaoDev44 — APPROVED** [review]

**Pull request merged**